### PR TITLE
Core protocol v3.0 - chunk memory layouts

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR adds draft content for the core protocol v3.0 section on memory layouts.

### CODE REVIEW

File: docs/protocol/core/v3.0.rst [hunk 1]
```diff
@@ -112,7 +112,48 @@ TODO define regular chunk grids, including how to form a key for each chunk in a
...
+
+In this memory layout, the binary values of the array elements are
+organised into a sequence such that the last dimension of the array is
+the fastest changing dimension, also known as "row-major" order. This
+layout is only applicable to arrays with fixed size data types.
+
+For example, for a two-dimensional array with chunk shape (dx, dy),
+the binary values for a given chunk are taken from chunk elements in
+the order (0, 0), (0, 1), (0, 2), ..., (dx - 1, dy - 3), (dx - 1, dy -
+2), (dx - 1, dy - 1).
```
Comment (martindurant) [inline docs/protocol/core/v3.0.rst]: Maybe worth mentioning here, that the memory offset/location within the chunk is determined by this ordering *and* the size of the dtype. Will endianness be in the dtype section only (I know it's mentioned already). That could be be loosely considered memory layout, especially for sub-byte types.

File: docs/protocol/core/v3.0.rst [hunk 2]
```diff
@@ -112,7 +112,48 @@ TODO define regular chunk grids, including how to form a key for each chunk in a
...
+
+F contiguous memory layout
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this memory layout, the binary values of the array elements are
+organised into a sequence such that the first dimension of the array
+is the fastest changing dimension, also known as "column-major" or
+"Fortran" order. This layout is only applicable to arrays with fixed
```
Comment (martindurant) [inline docs/protocol/core/v3.0.rst]: It is made specific what "F" stands for here. Either could add that "C" comes from the programming language of that name, or simply not mention Fortran at all.

**Comment by alimanfoo** [comment]
In the interests of having content together in one place, I'd like to merge this PR into the core-protocol-v3.0-dev branch. We can still discuss, revise and revisit anything after merge. I'll merge tomorrow if no objections.

File: docs/protocol/core/v3.0.rst (hunk 1 above)
Comment (alimanfoo) [reply docs/protocol/core/v3.0.rst]: Sounds good. Could you suggest an edit?

File: docs/protocol/core/v3.0.rst (hunk 2 above)
Comment (alimanfoo) [reply docs/protocol/core/v3.0.rst]: Fair point, have removed "Fortran".

**Pull request merged**